### PR TITLE
[OCPBUGS-44709]: Clarifying HCP requirement

### PR DESCRIPTION
**Pull request by lahinson**
Version(s): 4.17+

Issue: https://issues.redhat.com/browse/OCPBUGS-44709

Link to docs preview:
- Requirements for hosted control planes: https://86276--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-prepare/hcp-requirements.html
- Requirements to deploy HCP on OpenShift Virtualization: https://86276--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-deploy/hcp-deploy-virt#hcp-virt-reqs_hcp-deploy-virt
- Prerequisites for AWS: https://86276--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-deploy/hcp-deploy-aws#hcp-aws-prepare_hcp-deploy-aws
- Prerequisites for bare metal: https://86276--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-deploy/hcp-deploy-bm#hcp-bm-prepare_hcp-deploy-bm
- Prerequisites for non-bare-metal agent machines: https://86276--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-deploy/hcp-deploy-non-bm#hcp-non-bm-prereqs_hcp-deploy-non-bm

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: hosted_control_planes/hcp-prepare/hcp-requirements.adoc [hunk 1]
```diff
@@ -6,7 +6,9 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
-In the context of {hcp}, a _management cluster_ is an {product-title} cluster where the HyperShift Operator is deployed and where the control planes for hosted clusters are hosted. The management cluster and workers must run on the same infrastructure. For example, you cannot run your management cluster on bare metal and your workers on the cloud. However, the management cluster and workers do not need to run on the same platform. For example, you might run your management cluster on bare metal and workers on {VirtProductName}.
+In the context of {hcp}, a _management cluster_ is an {product-title} cluster where the HyperShift Operator is deployed and where the control planes for hosted clusters are hosted. 
+
+The management cluster and worker nodes must run on the same type of infrastructure. For example, you can run both the management cluster and the worker nodes on-premise, such as on a bare-metal platform or on {VirtProductName}. In addition, you can run both the management cluster and the worker nodes on cloud infrastructure, such as {aws-first}. However, you cannot run your management cluster on the cloud and your worker nodes on-premise, or vice versa.
```
[the remark this answers is not in the document]
Comment (lahinson) [reply hosted_control_planes/hcp-prepare/hcp-requirements.adoc:11]: Ignoring this suggestion. The SSG says that "bare-metal" is correct for the adjectival form of the phrase, which is what I'm using here.

File: modules/hcp-virt-reqs.adoc [hunk 2]
```diff
@@ -8,8 +8,8 @@
 
 As you prepare to deploy {hcp} on {VirtProductName}, consider the following information:
 
-* Run the hub cluster and workers on the same platform for {hcp}.
-* Each hosted cluster must have a cluster-wide unique name. A hosted cluster name cannot be the same as any existing managed cluster in order for {mce-short} to manage it.
+* The management cluster and workers must both run on {VirtProductName}.
```
Comment (zanetworker) [inline modules/hcp-virt-reqs.adoc:11]: The management cluster can't be on Virt, otherwise it would be nested virtualization which we don't support. It needs to be on Baremetal, especially if the management clsuter is where the VMs will be located (not virt external mode).

Comment (zanetworker) [reply modules/hcp-virt-reqs.adoc]: Makes sense for OpenShift Virtualization :)

**Review by skopacz1 — COMMENTED** [review]
Left one comment for your consideration, otherwise LGTM!

File: hosted_control_planes/hcp-prepare/hcp-requirements.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (lahinson) [reply hosted_control_planes/hcp-prepare/hcp-requirements.adoc:11]: @skopacz1 I'm tempted to leave the "vice versa" to avoid being too wordy. As an alternative, what do you think of replacing it with "or the other way around"?
`However, you cannot run your management cluster on the cloud and your worker nodes on-premise, or the other way around.`

[the remark this answers is not in the document]
Comment (skopacz1) [reply hosted_control_planes/hcp-prepare/hcp-requirements.adoc:11]: I think either leaving vice versa or using "the other way around" sounds fine, up to you! Would be good to remove vice versa but I'll leave it to your judgement as to which option sounds better for users.
